src/redux/data.ts

```diff
@@ -925,11 +928,15 @@ export const transactionsReceived = (
     }
     await dispatch(checkForUpdatedNonce(transactionData));
 
-    const { accountAddress, nativeCurrency, network } = getState().settings;
+    const { accountAddress, nativeCurrency } = getState().settings;
     const { purchaseTransactions } = getState().addCash;
     const { transactions } = getState().data;
     const { selected } = getState().wallets;
 
+    let { network } = getState().settings;
+    if (network === Network.mainnet && message?.meta?.chain_id) {
+      network = message?.meta?.chain_id;
```

Comment:
we cannot "set L2". This is the setting for testnets and L2 txs should be considered while the setting is used for mainnet.